feat(IMS): add ims images copy resource

houpeng80 opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:
  add ims images copy resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->

```release-note
  add ims images copy resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ims/' T
ESTARGS='-run TestAccImsImageCopy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims/ -v -run TestAccImsImageCopy_basic -timeout 360m -parallel 4
=== RUN   TestAccImsImageCopy_basic
=== PAUSE TestAccImsImageCopy_basic
=== CONT  TestAccImsImageCopy_basic
--- PASS: TestAccImsImageCopy_basic (672.14s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ims       672.193s
```

houpeng80 committed:
add ims images copy resource

houpeng80 committed:
add ims images copy resource

houpeng80 committed:
add ims images copy resource

Lance52259 committed:
fix(ecs/bind): Lock the instance ID when the related resources are binding (#2702)

* fix(ecs/bind): lock instance id when resources are binding

* test(ecs/bind): add new concurrency test for server group

* fix(ecs/detach): add delete check after volumes detach

ShiChangkuo committed:
docs: update the subcategory of CFW (#2712)

ShiChangkuo committed:
Update CHANGELOG for v1.44.2 (#2711)

ShiChangkuo committed:
Update APIs that resources depend-on v1.44.2

ShiChangkuo committed:
Update schema json files based-on v1.44.2

niuzhenguo commented on docs/resources/images_image_copy.md:
```diff
@@ -0,0 +1,56 @@
+variable "name" {}
+
+resource "huaweicloud_images_image_copy" "test" {
+  name = var.name
```
should at least provide requried parameters here like `image_id`

niuzhenguo commented on docs/resources/images_image_copy.md:
```diff
@@ -0,0 +1,56 @@
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `image_id` - (Required, String, ForceNew) Special the ID of the copied image.
```
s/Special/Specifies

niuzhenguo commented on huaweicloud/services/ims/resource_huaweicloud_images_image_copy.go:
```diff
@@ -0,0 +1,385 @@
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Special the ID of the copied image.`,
```
s/Special/Specifies.

niuzhenguo commented on huaweicloud/services/ims/resource_huaweicloud_images_image_copy.go:
```diff
@@ -0,0 +1,385 @@
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  `Specifies the name of the image.`,
```
better to also mention the length restriction to parameter description.

niuzhenguo commented on huaweicloud/services/ims/resource_huaweicloud_images_image_copy.go:
```diff
@@ -0,0 +1,385 @@
+		}
+
+		status := utils.PathSearch("status", getJobStatusRespBody, nil)
+		if err != nil {
```
unneeded err check here.

niuzhenguo commented on huaweicloud/services/ims/resource_huaweicloud_images_image_copy.go:
```diff
@@ -0,0 +1,385 @@
+			return nil, "", fmt.Errorf("creating ImsImageCopy job run fail")
+		}
+		entities := utils.PathSearch("entities", getJobStatusRespBody, nil)
+		if err != nil {
```
ditto.

ShiChangkuo committed:
docs: fix HCL syntax issue for examples (#2713)

Jason-Zhang9309 committed:
test(ecs): merge test cases of tags and disks into basic (#2717)

Lance52259 committed:
feat(apig/authorizer): refactor authorizer resource (#2640)

Lance52259 committed:
refactor(apig): refactor api resource (#2666)

Jason-Zhang9309 committed:
feat(ecs): support data_disk encryption (#2720)

Jason-Zhang9309 committed:
feat(ecs): support description in ecs instance (#2722)

dependabot[bot] committed:
chore(deps): bump golang.org/x/text from 0.3.7 to 0.3.8 (#2728)

Bumps [golang.org/x/text](https://github.com/golang/text) from 0.3.7 to 0.3.8.
- [Release notes](https://github.com/golang/text/releases)
- [Commits](https://github.com/golang/text/compare/v0.3.7...v0.3.8)

---
updated-dependencies:
- dependency-name: golang.org/x/text
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Lance52259 committed:
docs(workspace/desktop): add the description of default value (#2730)

dependabot[bot] committed:
chore(deps): bump golang.org/x/net (#2731)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.0.0-20211112202133-69e39bad7dc2 to 0.7.0.
- [Release notes](https://github.com/golang/net/releases)
- [Commits](https://github.com/golang/net/commits/v0.7.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hmbig2 committed:
feat(CPH): add a new datasource of server flavors (#2716)

hmbig2 committed:
feat(KMS): add a new resource KMS grant (#2707)

ShiChangkuo committed:
feat(vpc): add flow log resource (#2714)

houpeng80 committed:
feat: add ddm engines data source (#2723)

Jason-Zhang9309 committed:
feat(feat): make charging_mode available for user (#2732)

Jason-Zhang9309 committed:
fix(dds): set err to 404 when the list of role or user is empty (#2733)

ShiChangkuo committed:
feat(iam): add password policy support (#2734)

Lance52259 committed:
docs(vip/associate): add description of resource quantity limit (#2740)

ShiChangkuo committed:
chore(ci): disable flag-parameter rule (#2745)

deer-hang committed:
feat: as support as-configuration data source (#2729)

deer-hang committed:
feat: as group support datasource (#2724)

Lance52259 committed:
feat(port/attach): add security groups parameter support (#2739)

Jason-Zhang9309 committed:
chore(cce): add an example of using cce with k8s provider (#2718)

Lance52259 committed:
feat(rms/policy): add new policy definition query data source (#2719)

Lance52259 committed:
feat(nat/tags): gateway resource support tags parameter (#2726)

Jason-Zhang9309 committed:
feat(cce): partly fix prePaid node pool (#2741)

Lance52259 committed:
feat(dnat/rule): support the port range parameter (#2742)

Lance52259 committed:
feat(apig/api): refactor policy structure and support backend auth parameters (#2747)

ShiChangkuo committed:
chore(docs): use dos2unix to format files and fix misspell (#2748)

* chore(docs): use dos2unix to format files

* chore(docs): fix misspell

houpeng80 committed:
feat: add ddm flavors data source (#2736)

hmbig2 committed:
feat(CPH): add a new resource to manage server (#2743)

Lance52259 committed:
fix(rms/definitions): adjust endpoint URL and remove the region parameter (#2749)

* fix(rms/endpoints): the rms endpoint is the global endpoint

* fix(rms/definitions): remove the region parameter

Lance52259 committed:
feat(rms/policy): add new policy assignment resource (#2721)

Lance52259 committed:
fix(css/flavors): fix the wrong name of the fileter parameters (#2750)

* fix(css/flavors): fix the wrong name of the fileter parameters

* refactor(css/flavors): remove the empty return error

deer-hang committed:
fix(docs): remove deprecated fields from markdown (#2753)

houpeng80 committed:
feat: add ddm instance resource (#2751)

deer-hang committed:
feat(WAF): support pci_3ds and pci_dss (#2752)

deer-hang committed:
fix(docs): markdown field type fix (#2754)

houpeng80 committed:
fix:  decrease rocketmq wait delay time (#2756)

ShiChangkuo committed:
Update CHANGELOG for v1.45.0 (#2758)

ShiChangkuo committed:
Update APIs that resources depend-on v1.45.0

ShiChangkuo committed:
Update schema json files based-on v1.45.0

Lance52259 committed:
fix(eps): fix the wrong enum value and document format (#2773)

* docs(eps): adjust document format

* fix(eps): fix the wrong enum value

ShiChangkuo committed:
fix(cdm): unset the default cluster version (#2767)

Lance52259 committed:
test(rms/precheck): skip test if the domain id is not set (#2774)

niuzhenguo committed:
fix(gaussdb): fix dnNum issue for opengauss (#2762)

niuzhenguo committed:
refactor(test): extract base resources for testing (#2759)

Lance52259 committed:
fix(fgs/function): add param when updating function created by mirror (#2770)

Lance52259 committed:
refactor(nat): refactor gateway resource (#2757)

Lance52259 committed:
refactor(nat/dnat): rewrite dnat rule resource (#2768)

niuzhenguo committed:
fix(test): use fixed cidr to fix test issues (#2781)

deer-hang committed:
fix(doc): waf dedicated domain markdown (#2780)

Jason-Zhang9309 committed:
feat(ces): update ces alarm rule with v2 api (#2776)

Lance52259 committed:
refactor(nat/snat): rewrite snat rule resource (#2772)

houpeng80 committed:
feat: kafka instance support prepaid (#2777)

deer-hang committed:
fix: waf dedicated instance support res_tenant (#2783)

Lance52259 committed:
refactor(nat/gateway): move the data source to the new dir (#2784)

* refactor(nat/gateway): move the data source to the new dir

* refactor(nat/gateway): refactoe functions and acc test

ShiChangkuo committed:
Update CHANGELOG for v1.45.1 (#2782)

ShiChangkuo committed:
Update APIs that resources depend-on v1.45.1

ShiChangkuo committed:
Update schema json files based-on v1.45.1

hmbig2 committed:
feat(CPH): add a new dataSource to get phone flavors (#2761)

houpeng80 committed:
feat: add ddm instance nodes data source (#2766)

houpeng80 committed:
feat(lb): add public ip attribute (#2786)

houpeng80 committed:
fix(test): make dds test happy (#2787)

houpeng80 committed:
feat: add ddm schema resource (#2764)

houpeng80 committed:
feat: add ddm instaces data source (#2765)

deer-hang committed:
feat: autoscaling support notification (#2771)

hmbig2 committed:
feat(CPH): add a new datasource to get phone images (#2769)

ShiChangkuo committed:
fix(test): fix Reference to undeclared resource issue (#2790)

following up 1c6dc94

deer-hang committed:
refactor(dns): remove value spec field (#2789)

Jason-Zhang9309 committed:
feat(cce): support storage in node pool (#2795)

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

niuzhenguo committed:
fix(test/rds): fix rds_account test issue (#2797)

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

Jason-Zhang9309 committed:
fix(ces): add default value for alarm_type (#2805)

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource

houpeng80 committed:
add ims image copy resource
